src/plots/geo/geo.js

```diff
@@ -136,21 +135,20 @@ proto.plot = function(geoData, fullLayout, promises) {
 };
 
 proto.onceTopojsonIsLoaded = function(geoData, geoLayout) {
-    var scattergeoData = [],
-        choroplethData = [];
+    var traceData = {};
 
     this.drawLayout(geoLayout);
 
     for(var i = 0; i < geoData.length; i++) {
         var trace = geoData[i];
-        var traceType = trace.type;
 
-        if(traceType === 'scattergeo') scattergeoData.push(trace);
-        else if(traceType === 'choropleth') choroplethData.push(trace);
+        traceData[trace.type] = traceData[trace.type] || [];
+        traceData[trace.type].push(trace);
     }
 
-    if(scattergeoData.length>0) plotScatterGeo.plot(this, scattergeoData);
-    if(choroplethData.length>0) plotChoropleth.plot(this, choroplethData, geoLayout);
+    for(var traceType in traceData){
```

Comment:
:cow2: we've been trying to avoid the `for(var i in obj)` pattern.

Either use `Object.keys(traceData)` or store an array of trace types along side the `traceData` object.